server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -715,47 +722,54 @@ Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, int num, Se
 
     long maxResultsSize = tableConfiguration.getAsBytes(Property.TABLE_SCAN_MAXMEM);
 
-    if (columns.size() == 0) {
-      iter.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
-    } else {
-      iter.seek(range, LocalityGroupUtil.families(columns), true);
-    }
-
     Key continueKey = null;
     boolean skipContinueKey = false;
 
     boolean endOfTabletReached = false;
-    while (iter.hasTop()) {
 
-      value = iter.getTopValue();
-      key = iter.getTopKey();
+    try {
+      if (columns.size() == 0) {
+        iter.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
```

Comment:
> The key placed in the ScanYieldException is not the start of the initial range. It is the key where it left off in the underlying scan

You can't guarantee this. What I am saying is: what _if_ the key you see is the same?

Your last comment makes me think that you intend for this exception to never be thrown when "no progress" was made. If that's correct, documentation needs to adequately warn..